### DISCUSSION TITLE
[OPENSSL] Can I call Session.invalidate() of ongoing SSL connection to disable caching?

### DISCUSSION THREAD

**Discussion by tamilselvan-rv-12109** [netty/netty#14151] [Q&A] [answered]
[using tc-native OpenSSL] I have a http client which connects to multiple hosts. I have enabled SSL session caching for my client. I don't want to allow SSL session caching for certain hosts. So, I am invalidating the session immediately after the SSL handshake is completed for those hosts. Here there is an active Connection using that session is still alive and transferring data. Does this affect the current SSL connection? If it affects, is there any right way to do it better?

**Comment by chrisvest** [comment, marked as answer]
Invalidating an SSL session only means it can't be reused for subsequent connections. The existing session on the current connection is not impacted.

Normally you'd use this to deliver a higher-level error for failed authentication, like HTTP 401 or 403, or a gRPC error, etc. But it will also work for your use case, if I understand it right.

**Reply by tamilselvan-rv-12109** [reply]
Thanks for you answer.
